### ISSUE TITLE
More complex markdown link and aliased link is not processed properly

### ISSUE THREAD

**Issue by spacegiant** [obsidian-html/obsidian-html#449]
# Expected behaviour
A links such as `[[Folder Name/Sub Folder/My Note]]` and `[[My long note name|Note name]]` should have their pages output like all the others.

# Actual behaviour
A link such as `[[Folder Name/Sub Folder/My Note]]` and `[[My long note name|Note name]]` ends up as `not_created.html`

# Notes
- Built on MacBook Air M1
- Deployed to Github Pages

**Comment by dwrolvink** [comment]
- What is special about the links that you shared?
- Do you have links that do work?
- What version of obsidianhtml are you using?
- What do you get when you run `obsidianhtml version`?

I've created a link like those below, and they work.
``` md
[[Demonstrations/Test pages/proper markdown]]
[[proper markdown|test]]
[[Demonstrations/Test pages/proper markdown|test]]
[[Demonstrations/Test pages/proper markdown |test]]
```

These all work in Obsidian and in the html output running the latest master version, except for the last link, which works in Obsidian but not in the html output.

That is because of the space, but I can't use that to explain your examples, because they don't have spaces around the `|`.

Could you set up a minimum setup to recreate this? 
For example, creating a vault with this content:
```
noteA.md
folder
  noteB.md
  note C.md
``` 

And the contents of noteA.md being:
``` md
[[folder/noteB]]
[[folder/noteB|Bee]]
[[folder/note C]]
[[folder/note C|Cee]]
``` 

Which links then don't work in the html output? (Assuming they all work in Obsidian).

I'm asking for this so that I can get to a point where I can recreate the bug, because as it stands I have no idea what problem you are facing here.

**Comment by spacegiant** [comment]
Thanks for replying so quickly. I should have put more info but I didn't know how to do that with pip. But I learnt. Here's some more detailed info:

- `pip list` tells me I'm using 3.2.3 of obsidian-html. I just updated and tried again.
- `obsidianhtml version` corresponds with 3.2.3
- The link I am having a problem is `[[Solo Notes/Tricube Cyberpunk/web/Setup]]`
- The link works in Obsidian but not in the output HTML site.
- Other links are all working fine.
- Now I've looked at it again, I think the reason for this is because I have another file elsewhere already called `Setup`. So instead of just allowing [[Setup]] for the link, Obsidian requires the full path. `[[Solo Notes/Tricube Cyberpunk/web/Setup]]`
- The version with the alias I can't reproduce. That was a human error at my end. Apologies.

I hope that makes sense.

I should also add that obsidian-html is absolutely brilliant! Thanks for making it available. :D

**Comment by dwrolvink** [comment]
I have recreated the folder structure in my root: `/Solo Notes/Tricube Cyberpunk/web/Setup`, `/Setup`, and then linked to it with `[[Solo Notes/Tricube Cyberpunk/web/Setup]]` from a note in the root and from another folder. It all works for me.

Could you **share your config.yml** and what commands you have used to create the html from your vault?  

I am wondering whether you have excluded the Solo Notes folder somehow.
Could you confirm whether or not the file `{html_output_dir}/Solo Notes/Tricube Cyberpunk/web/Setup.html` is generated at all?

If it is not, then a quick tip would be to look whether:

- You have configured any folders to be excluded.
- You have configured a subset of your vault instead of the whole of it (see https://obsidian-html.github.io/Configurations/Configuration%20Options.html#!obsidian_folder_path_str)
- Try setting the following in your config.yml and then do the check again:
    ``` yaml
    toggles:
      # If this is false only the entrypoint and all its links (recursively) are processed
      # if true all the notes will be processed
      process_all: True
    ```

If this all does not work then I am at the end of my troubleshooting rope for now.
If possible, uploading your vault to a git repo so that I can use it to compile the html would be extremely helpful.

ps. And no problem, glad you like it. I've been using FOSS my entire life, just happy to be able to contribute something back.

**Comment by spacegiant** [comment]
I've managed to recreate the issue on a small example Obsidian vault to share, which I attach here...
[obsidian-test.zip](https://github.com/obsidian-html/obsidian-html/files/9893432/obsidian-test.zip)
This includes the output html website with link to uncreated page.

My actual config.yml for reference. Had to add .txt to upload here.
[config.yml.txt](https://github.com/obsidian-html/obsidian-html/files/9893416/config.yml.txt)

The command I used to run the process is `obsidianhtml convert -i config.yml `
I run this from a folder on my desktop. Obsidian vaults live elsewhere.

You'll see that I'm exporting a subset of the vault, but all files are in that vault.  

I tested the website locally using [http-server](https://github.com/http-party/http-server)

**Comment by dwrolvink** [comment]
Could you please add the example vault, now I only have the output, I need to compare the input with the config with the output.

**Comment by dwrolvink** [comment]
Oh I think I see what the problem is.

You have configured:
```
obsidian_folder_path_str: "/Users/craig/Desktop/ObsidianTest/My Test/My Folder
```

This will make that path `/` in "relative to root" terms.

Then you have the link: `[My Folder/Setup](My Folder/Setup)`
Obsidianhtml will then look for a folder called `My Folder`. This will fail, because your configured "vault" has no folder with that name.

See also the first line in this block, it is more important than it might seem at first:
``` yaml
# The first folder that contains all obsidian files
# Can be absent when toggles/compile_md == False
# Use full path or relative path, but don't use ~/
obsidian_folder_path_str: '<REQUIRED_INPUT>'
```
I should maybe add a warning when the given folder is not the vault root, as I have that already built into `obsidianhtml run`.

---

tl;dr: 
- Configure `obsidian_folder_path_str` to be your vault root (this is the only supported configuration)
- Exclude any folders that you don't want in your output _(by default only the notes that are linked to (in)directly by your entrypoint node are included.)_

**Comment by spacegiant** [comment]
That fixed it! Thanks :D

And just in case, I attach the Test Obsidian Vault...
[My Test.zip](https://github.com/obsidian-html/obsidian-html/files/9893564/My.Test.zip)

Do you want me to close this?

**Comment by dwrolvink** [comment]
Good!

I'll close it. 
The cause of your issue will be addressed in #451 you can keep an eye on that if you're interested.

Good luck with your note taking!

**Comment by spacegiant** [comment]
Thank you!

**Comment by dwrolvink** [comment]
https://obsidian-html.github.io/Configurations/Modes/Filtering%20notes.html fyi

**Comment by spacegiant** [comment]
Nice! Thanks for the update.
